src/layout.rs

```diff
@@ -0,0 +1,295 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANtIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use std::iter::Iterator;
+
+use hashbrown::{HashMap, HashSet};
+
+use pyo3::prelude::*;
+
+use petgraph::graph::NodeIndex;
+use petgraph::prelude::*;
+use petgraph::EdgeType;
+
+type Nt = f64;
+pub type Point = [Nt; 2];
+type Graph<Ty> = StableGraph<PyObject, PyObject, Ty>;
+
+const LBOUND: Nt = 1e-8;
+
+#[inline]
+fn l2norm(x: Point) -> Nt {
+    (x[0] * x[0] + x[1] * x[1]).sqrt()
+}
+
+#[inline]
+fn max(x: Nt, y: Nt) -> Nt {
+    if x > y {
+        x
+    } else {
+        y
+    }
+}
```

Comment:
Is there an advantage to defining a custom inline function over just calling `x.max(y)` where we need it?

Reply:
no, i removed it.